scripts/deploy-staking.ts

```diff
@@ -17,12 +17,7 @@ async function main() {
   const StakingContract = await ethers.getContractFactory('Staking', owner)
   const staking: Staking = await StakingContract.deploy(values.NEW_TOKEN, values.TREASURY)
 
-  console.info(`Check tx here: ${getEtherscanUri('tx', staking.deployTransaction.hash)}`)
-
-  await staking.deployed()
```

Comment:
How do you deploy if this is removed?

Reply:
.deployed() is only used to wait for the deployment. The contract is actually deployed by calling the method .deploy(), which is already there. 

Also, this .deployed() method is not available in the new versions that why I removed it. Now I have added the alternate which is .waitForDeployment()